src/mappings/index.ts

```diff
@@ -32,16 +32,17 @@ import {
   validateInteraction,
 } from './utils/consolidator'
 import { emoteId, ensure, ensureInteraction, isEmpty, eventId } from './utils/helper'
-import { System } from '../types'
+import { SystemRemarkCall  } from '../types/calls'
 import { Context } from './utils/types'
 import logger, { logError } from './utils/logger'
 import { create, get } from './utils/entity'
 import { fetchMetadata } from './utils/metadata'
-import { DatabaseManager } from '@subsquid/hydra-common'
+import { Store} from '@subsquid/substrate-processor'
 import {updateCache} from './utils/cache'
 
+
 export async function handleRemark(context: Context): Promise<void> {
-  const remark = new System.RemarkCall(context.extrinsic).remark
+  const remark = new SystemRemarkCall(context).asV1020.remark
```

Comment:
Wouldn't be this issue for older blocks?

Reply:
Nah, that's only one version of remark call spec in Runtime. Btw, new squid supports type safe types, you can check this [here](https://github.com/subsquid/squid-template/blob/main/src/processor.ts#L63)

Reply:
Perfect!